Functions/Send-SyslogMessage.ps1

```diff
@@ -73,185 +74,383 @@ function Send-SyslogMessage
+        [Parameter(Mandatory = $false,
+                    ValueFromPipeline = $false)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet('UDP','TCP')]
+        [String]
+        $Transport = 'UDP',
+
+        # Framing method used for the message, default is 'Octet-Counting' (see RFC6587 section 3.4). This only applies when TCP is used for transport (no effect on UDP messages).
+        [Parameter(Mandatory = $false,
+                    ValueFromPipeline = $false)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet('Octet-Counting','Non-Transparent-Framing','None')]
+        [String]
+        $FramingMethod='Octet-Counting',
 	
         #Send an RFC3164 fomatted message instead of RFC5424.
-        [Parameter(mandatory = $True,
-                   ParameterSetName = 'RFC3164')]
+        [Parameter(Mandatory = $True,
+                    ValueFromPipeline = $true,
+                    ParameterSetName = 'RFC3164')]
         [switch]
-        $RFC3164
+        $RFC3164,
+
+        #Enable static source IP detection for strict adherence to RFC 5424 (section 6.2.4), if FQDN is not available. Using this might reduce performance in high-volume scenarios.
+        [Parameter(Mandatory = $false,
+                    ValueFromPipeline = $false)]
+        [switch]
+        $DetectSourceIP
     )
+    Begin 
+    {
+        Write-Verbose "Starting the BEGIN block..."
+
+        # Create an ASCII Encoding object
+        $Encoding = [System.Text.Encoding]::ASCII
+
+        if ($Hostname -eq '')
+        {            
+            Write-Verbose "No HOSTNAME value was provided, so automatic selection will now be attempted..."
+            <#
+            According to RFC 5424 (section 6.2.4), we need to send our HOSTNAME field as one of these 5 (in order of priority)
+            1.  FQDN
+            2.  Static IP address
+            3.  Hostname - Windows always has one of these, so this is our last resort
+            4.  Dynamic IP address - We will never get to this one
+            5.  the NILVALUE - or this one
 
-    # Evaluate the facility and severity based on the enum types
-    $Facility_Number = $Facility.value__
-    $Severity_Number = $Severity.value__
-    Write-Verbose -Message "Syslog Facility, $Facility_Number, Severity is $Severity_Number"
+            Windows should always, in the worst case, have a result at 3, the hostname or computer name from which this command is run.
+            #>        
+            
+            # Get the IP global proprties of the local machine, to determine if an FQDN is available
+            #$ComputerName = [System.Net.Dns]::GetHostByName($Env:COMPUTERNAME).HostName
+            #([System.Net.NetworkInformation.IPGlobalProperties]::GetIPGlobalProperties().Hostname) + '.' + ([System.Net.NetworkInformation.IPGlobalProperties]::GetIPGlobalProperties().DomainName)
+            $IpGlobalProps = [System.Net.NetworkInformation.IPGlobalProperties]::GetIPGlobalProperties()
```

Comment:
What are the advantages of using [System.Net.NetworkInformation.IPGlobalProperties]::GetIPGlobalProperties() over the Environment Variable?